window/src/os/windows/window.rs

```diff
@@ -683,10 +683,15 @@ impl WindowOps for Window {
         )
     }
 
-    fn set_clipboard(&self, _clipboard: Clipboard, text: String) -> Future<()> {
-        Future::result(
-            clipboard_win::set_clipboard_string(&text).context("Error setting clipboard"),
-        )
+    fn set_clipboard(&self, clipboard: Clipboard, text: String) -> Future<()> {
```

Comment:
How about introducing a new configuration option named something like `treat_primary_selection_as_clipboard_on_windows` that controls this?
I want the default to be be `true`, but it would allow you to set it to false
to get the behavior that you want.

You can add the option in a similar way to this:
https://github.com/wez/wezterm/blob/main/config/src/lib.rs#L666-L667

Then to access the config, something like this:

```rust
fn set_clipboard(&self, clipboard: Clipboard, text: String) -> Future<()> {
    Connection::with_window_inner(self.0, move |inner| {
        if matches!(clipboard, Clipboard::Clipboard) || inner.config.treat_primary_selection_as_clipboard_on_windows {
         clipboard_win::set_clipboard_string(&text).context("Error setting clipboard"),
       } else {
         Ok(())
       }
    })
}

               
```

Reply:
I think this is redundant. There's already a property https://wezfurlong.org/wezterm/config/lua/keyassignment/CopyTo.html to set the copy destination, if someone would like selection automatically copied they can just set the destination to be `Clipboard` or `ClipboardAndPrimarySelection`. And if you want this behavior to be default, you can make the mousebinding default to these value as well.
Since the `PrimarySelection` is defined and only supported on the X platform (probably wayland i'm not sure), it's technically correct to ignore this operation on unsupported platform, and, if copy to clipboard is expected, set it (or default) to clipboard is more intuitive.